userspace/engine/falco_engine.cpp

```diff
@@ -91,7 +91,7 @@ void falco_engine::list_fields(std::string &source, bool verbose, bool names_onl
 	// applicable to multiple event sources.
 	for(auto &it : m_filter_factories)
 	{
-		if(source != "" && source != it.first)
+		if(source != "all" && source != it.first)
```

Comment:
Honestly, I don't like the idea to introduce a kind of textual constant (i.e., `all`). I would prefer the empty string as before.

Reply:
Sure, I was on the fence. Switched it back.